GoogleChromeLabs/ps-analysis-tool
# Feature: Publish packages on the NPM registry

## Description
This PR aims to publish the following packages on NPM registry:
- `analysis-utils`
- `cli-dashboard`
- `cli`
- `common`
- `design-system`
- `i18n`
- `library-detection`

## Relevant Technical Choices

- Script for publishing packages locally and in the npm registry has been added. Also, the script for unpublishing the locally published package has been added.
- The CLI package generates a JSON file and an HTML file. The HTML file can be viewed without running a dev server.
- Once the CLI is done with site/sitemap analysis the index.tml file from cli-dashboard is read and then we add the JSON data under a script tag in the window object associated with PSAT_DATA.
- Then this index.html file along with the cliDashboard.js file is copied to the out directory.
- To make report generation work we add the script for report generation in the index.html that is read from the cli-dashboard file as well in a base64 format under the window object with a variable PSAT_REPORT.
- Once the user clicks on the download report the window.PSAT_REPORT is read then it is base64 decoded which gives the report javascript, now a new index.html file is generated with the report javascript under the script tag which is the static HTML format of the cookie landing page.
- New commands have been added that utilise the workspace feature of NPM in the following format: `packagename:build`


## Testing Instructions
This PR has to be tested in 2 different ways:
1) When `PSAT` is installed in a project:
- Clone this branch and run `npm i`.
- (Optional) the registry start command required `coreutils` which is not available on OSX so its better to install coreutils using `brew install coreutils`.
- In the terminal run `npm run local-registry:start`.
- In the terminal run `npm run publish:all:local` this will build all the packages that need to be published.
- Check the cli using `npm run cli` command and check the extension if it works properly.
- Then go to `http://localhost:4873` you will see all the packages that have been published.
- Now create a new directory outside ps-analysis-tool.
- In the new directory run `npm init`.
- Then run `npm install @ps-analysis-tool/cli --registry=http://localhost:4873`.
- Now run the cli using the `npx psat` command.
- Verify that the cli now runs properly.
- Now in terminal where you ran `npm run local-registry:start` run `npm run local-registry:stop`.

2) When `PSAT` is installed globally:
- Install the psat globally using `npm i @ps-analysis-tool/cli @ps-analysis-tool/cli-dashboard -g --registry=http://localhost:4873`.
- Run the cli using the `psat` command.

## Screenshot/Screencast


---

## Checklist






Fixes #589
